src/main/java/com/google/api/codegen/transformer/py/PythonModelTypeNameConverter.java

```diff
@@ -0,0 +1,245 @@
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer.py;
+
+import com.google.api.codegen.config.FieldConfig;
+import com.google.api.codegen.transformer.ModelTypeNameConverter;
+import com.google.api.codegen.util.Name;
+import com.google.api.codegen.util.TypeName;
+import com.google.api.codegen.util.TypeNameConverter;
+import com.google.api.codegen.util.TypedValue;
+import com.google.api.codegen.util.py.PythonTypeTable;
+import com.google.api.tools.framework.model.EnumType;
+import com.google.api.tools.framework.model.EnumValue;
+import com.google.api.tools.framework.model.ProtoElement;
+import com.google.api.tools.framework.model.TypeRef;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.protobuf.DescriptorProtos.FieldDescriptorProto.Type;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class PythonModelTypeNameConverter implements ModelTypeNameConverter {
+  private static final String GOOGLE_PREFIX = "google";
+
+  private static final String GOOGLE_CLOUD_PREFIX = GOOGLE_PREFIX + ".cloud";
+
+  private static final String GOOGLE_CLOUD_PROTO_PREFIX = GOOGLE_CLOUD_PREFIX + ".proto";
+
+  /** A map from primitive type to its corresponding Python types */
+  private static final Map<Type, String> PRIMITIVE_TYPE_MAP =
+      ImmutableMap.<Type, String>builder()
+          .put(Type.TYPE_DOUBLE, "float")
+          .put(Type.TYPE_FLOAT, "float")
+          .put(Type.TYPE_INT32, "int")
+          .put(Type.TYPE_INT64, "long")
+          .put(Type.TYPE_UINT32, "int")
+          .put(Type.TYPE_UINT64, "long")
+          .put(Type.TYPE_SINT32, "int")
+          .put(Type.TYPE_SINT64, "long")
+          .put(Type.TYPE_FIXED32, "int")
+          .put(Type.TYPE_FIXED64, "long")
+          .put(Type.TYPE_SFIXED32, "int")
+          .put(Type.TYPE_SFIXED64, "long")
+          .put(Type.TYPE_BOOL, "bool")
+          .put(Type.TYPE_STRING, "string")
+          .put(Type.TYPE_BYTES, "bytes")
+          .build();
+
+  /** A map from primitive types to its default value. */
+  private static final Map<Type, String> PRIMITIVE_ZERO_VALUE =
+      ImmutableMap.<Type, String>builder()
+          .put(Type.TYPE_BOOL, "False")
+          .put(Type.TYPE_DOUBLE, "0.0")
+          .put(Type.TYPE_FLOAT, "0.0")
+          .put(Type.TYPE_INT64, "0")
+          .put(Type.TYPE_UINT64, "0")
+          .put(Type.TYPE_SINT64, "0")
+          .put(Type.TYPE_FIXED64, "0")
+          .put(Type.TYPE_SFIXED64, "0")
+          .put(Type.TYPE_INT32, "0")
+          .put(Type.TYPE_UINT32, "0")
+          .put(Type.TYPE_SINT32, "0")
+          .put(Type.TYPE_FIXED32, "0")
+          .put(Type.TYPE_SFIXED32, "0")
+          .put(Type.TYPE_STRING, "\'\'")
+          .put(Type.TYPE_BYTES, "b\'\'")
+          .build();
+
+  // TODO (geigerj): Read this from configuration?
```

Comment:
This will be driven from `artman` via configuration in `googleapis`. There is actually [some configuration in `googleapis` already for `googleapis-common-protos`](https://github.com/googleapis/googleapis/blob/master/gapic/packaging/common_protos.yaml), but I think this is actually somewhat different.

This list is not actually "common protos" per se (e.g., IAM is not in `googleapis-common-protos`), but rather a list of proto package names we don't modify to add `cloud` in the path. The packages contained in `googleapis-common-protos` is a subset of this list.

I don't think this needs to be addressed in this PR. Creating https://github.com/googleapis/toolkit/issues/1113 to track.